### PR TITLE
Dịch "adagrad_vn" - Phần 5

### LINKED ISSUE #1943: Dịch "adagrad_vn" - Phần 5
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/adagrad_vn.md#L299-L399

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by trGiang99**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Điền số issue sau dấu `#` (không có dấu cách): Close #1943

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn. Bạn đã kiểm tra rằng mỗi đoạn văn đã nằm một dòng?

* [x] Điền tên đầy đủ của bạn vào mục **Những người thực hiện** ở cuối tập tin.

* [ ] Điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Lưu ý cho người phản biện (Reviewer)

1. Chỉ review nếu Pull Request (PR) này đã sẵn sàng cho việc review. Một PR sẵn sàng khi PR đó không có label `status: wip`.
2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

### CODE REVIEW

File: chapter_optimization/adagrad_vn.md [hunk 1]
```diff
@@ -302,13 +302,13 @@ d2l.show_trace_2d(f_2d, d2l.train_2d(adagrad_2d))
 ## Implementation from Scratch
 -->
 
-## *dịch tiêu đề phía trên*
+## Lập trình lại từ đầu
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:305]:
```suggestion
## Lập trình từ đầu
```

File: chapter_optimization/adagrad_vn.md [hunk 2]
```diff
@@ -344,13 +344,13 @@ d2l.train_ch11(adagrad, init_adagrad_states(feature_dim),
 ## Concise Implementation
 -->
 
-## *dịch tiêu đề phía trên*
+## Cách lập trình Ngắn gọn
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:347]:
```suggestion
## Lập trình Súc tích
```

File: chapter_optimization/adagrad_vn.md [hunk 3]
```diff
@@ -344,13 +344,13 @@ d2l.train_ch11(adagrad, init_adagrad_states(feature_dim),
...
 
 <!--
 Using the `Trainer` instance of the algorithm `adagrad`, we can invoke the Adagrad algorithm in Gluon.
 -->
 
-*dịch đoạn phía trên*
+Sử dụng đối tượng `Trainer` trong thuật toán `adagrad`, ta có thể gọi đến thuật toán Adagrad trong Gluon.
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:353]:
```suggestion
Sử dụng đối tượng `Trainer` trong thuật toán `adagrad`, ta có thể gọi thuật toán Adagrad trong Gluon.
```

File: chapter_optimization/adagrad_vn.md [hunk 4]
```diff
@@ -373,7 +373,12 @@ d2l.train_gluon_ch11('adagrad', {'learning_rate': 0.1}, data_iter)
 * On deep learning problems Adagrad can sometimes be too aggressive in reducing learning rates. We will discuss strategies for mitigating this in the context of :numref:`sec_adam`.
 -->
 
-*dịch đoạn phía trên*
+* Adagrad liên tục giảm giá trị của tốc độ học theo từng toạ độ.
+* Thuật toán sử dụng độ lớn của gradient như một phương thức để điều chỉnh tiến độ thực hiện - các toạ độ với gradient lớn được cân bằng bởi tốc độ học nhỏ.
+* Ta thường không thể tính toán chính xác được đạo hàm bậc hai trong các bài toán học sâu do hạn chế về bộ nhớ và khả năng tính toán. Gradient có thể trở thành một biến đại diện hữu ích.
+* Nếu bài toán tối ưu có cấu trúc không được đồng đều, Adagrad có thể làm giảm bớt sự biến dạng đó.
+* Adagrad thường khá hiệu quả đối với các đặc trưng thưa, khi việc giảm tốc độ học cần phải chậm lại tại các thời điểm đặc trưng đó hiếm khi xảy ra.
+* Trong các bài toán học sâu, Adagrad đôi khi làm giảm tốc độ học quá mạnh. Ta sẽ thảo luận các chiến lược nhằm hạn chế vấn đề này trong ngữ cảnh :numref:`sec_adam`.
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:381]:
```suggestion
* Trong các bài toán học sâu, Adagrad đôi khi làm giảm tốc độ học quá mạnh. Ta sẽ thảo luận các chiến lược nhằm giảm bớt vấn đề này trong ngữ cảnh :numref:`sec_adam`.
```

File: chapter_optimization/adagrad_vn.md [hunk 5]
```diff
@@ -373,7 +373,12 @@ d2l.train_gluon_ch11('adagrad', {'learning_rate': 0.1}, data_iter)
 * On deep learning problems Adagrad can sometimes be too aggressive in reducing learning rates. We will discuss strategies for mitigating this in the context of :numref:`sec_adam`.
 -->
 
-*dịch đoạn phía trên*
+* Adagrad liên tục giảm giá trị của tốc độ học theo từng toạ độ.
+* Thuật toán sử dụng độ lớn của gradient như một phương thức để điều chỉnh tiến độ thực hiện - các toạ độ với gradient lớn được cân bằng bởi tốc độ học nhỏ.
+* Ta thường không thể tính toán chính xác được đạo hàm bậc hai trong các bài toán học sâu do hạn chế về bộ nhớ và khả năng tính toán. Gradient có thể trở thành một biến đại diện hữu ích.
+* Nếu bài toán tối ưu có cấu trúc không được đồng đều, Adagrad có thể làm giảm bớt sự biến dạng đó.
+* Adagrad thường khá hiệu quả đối với các đặc trưng thưa, khi việc giảm tốc độ học cần phải chậm lại tại các thời điểm đặc trưng đó hiếm khi xảy ra.
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:380]:
```suggestion
* Adagrad thường khá hiệu quả đối với các đặc trưng thưa, trong đó tốc độ học cần giảm chậm hơn cho các tham số hiếm khi xảy ra.
```

Comment (davidnvq) [reply chapter_optimization/adagrad_vn.md:380]: a nghĩ  term ở đây ám chỉ tới các phần tử trong parameters mà ứng với các giá trị thưa trong đặc trưng.

File: chapter_optimization/adagrad_vn.md [hunk 6]
```diff
@@ -393,7 +398,15 @@ a matrix $\mathbf{M}$ satisfy $|\lambda_i - \mathbf{M}_{jj}| \leq \sum_{k \neq j
 6. How would you need to modify Adagrad to achieve a less aggressive decay in learning rate?
 -->
 
-*dịch đoạn phía trên*
+1. Chứng minh rằng một ma trận trực giao $\mathbf{U}$ và một vector $\mathbf{c}$ thoả mãn điều kiện: $\|\mathbf{c} - \mathbf{\delta}\|_2 = \|\mathbf{U} \mathbf{c} - \mathbf{U} \mathbf{\delta}\|_2$.
+Tại sao biểu thức trên lại biểu thị rằng độ nhiễu loạn không thay đổi khi biến đổi trực giao các biến?
+2. Thử áp dụng Adagrad đối với $f(\mathbf{x}) = 0.1 x_1^2 + 2 x_2^2$ và đối với hàm mục tiêu được quay 45 độ,
+tức là $f(\mathbf{x}) = 0.1 (x_1 + x_2)^2 + 2 (x_1 - x_2)^2$. Adagrad có vận hành khác đi hay không?
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:404]:
```suggestion
tức là $f(\mathbf{x}) = 0.1 (x_1 + x_2)^2 + 2 (x_1 - x_2)^2$. Adagrad có hoạt động khác đi hay không?
```

File: chapter_optimization/adagrad_vn.md [hunk 7]
```diff
@@ -393,7 +398,15 @@ a matrix $\mathbf{M}$ satisfy $|\lambda_i - \mathbf{M}_{jj}| \leq \sum_{k \neq j
...
+3. Chứng minh [Định lý Gerschgorin](https://en.wikipedia.org/wiki/Gershgorin_circle_theorem), định lý phát biểu rằng với các trị riêng $\lambda_i$ của
+ma trận $\mathbf{M}$, tồn tại $j$ thoả mãn $|\lambda_i - \mathbf{M}_{jj}| \leq \sum_{k \neq j} |\mathbf{M}_{jk}|$.
+4. Từ định lý Gerschgorin, ta có thể chỉ ra điều gì về các trị riêng của ma trận đường chéo tiền điều kiện (*diagonally preconditioned matrix*) $\mathrm{diag}^{-\frac{1}{2}}(\mathbf{M}) \mathbf{M} \mathrm{diag}^{-\frac{1}{2}}(\mathbf{M})$?
+5. Hãy thử áp dụng Adagrad cho một mạng thực sự sâu như :numref:`sec_lenet` khi sử dụng Fashion MNIST.
+6. Bạn sẽ thay đổi Adagrad như thế nào để tốc độ học không còn bị suy hao quá mạnh?
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:409]:
```suggestion
6. Bạn sẽ thay đổi Adagrad như thế nào để tốc độ học không suy giảm quá mạnh?
```

File: chapter_optimization/adagrad_vn.md [hunk 8]
```diff
@@ -373,7 +373,12 @@ d2l.train_gluon_ch11('adagrad', {'learning_rate': 0.1}, data_iter)
 * On deep learning problems Adagrad can sometimes be too aggressive in reducing learning rates. We will discuss strategies for mitigating this in the context of :numref:`sec_adam`.
 -->
 
-*dịch đoạn phía trên*
+* Adagrad liên tục giảm giá trị của tốc độ học theo từng toạ độ.
+* Thuật toán sử dụng độ lớn của gradient như một phương thức để điều chỉnh tiến độ thực hiện - các toạ độ với gradient lớn được cân bằng bởi tốc độ học nhỏ.
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:377]:
```suggestion
* Thuật toán sử dụng độ lớn của gradient như một phương thức để điều chỉnh tiến độ học - các toạ độ với gradient lớn được cân bằng bởi tốc độ học nhỏ.
```

File: chapter_optimization/adagrad_vn.md [hunk 9]
```diff
@@ -373,7 +373,12 @@ d2l.train_gluon_ch11('adagrad', {'learning_rate': 0.1}, data_iter)
...
+* Ta thường không thể tính toán chính xác được đạo hàm bậc hai trong các bài toán học sâu do hạn chế về bộ nhớ và khả năng tính toán. Gradient có thể trở thành một biến đại diện hữu ích.
```
Comment (davidnvq) [inline chapter_optimization/adagrad_vn.md:378]:
```suggestion
* Tính chính xác đạo hàm bậc hai thường không khả thi trong các bài toán học sâu do hạn chế về bộ nhớ và khả năng tính toán. Gradient có thể trở thành một biến đại diện hữu ích.
```

File: chapter_optimization/adagrad_vn.md [hunk 10]
```diff
@@ -431,4 +444,4 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 * 
 
 <!-- Phần 5 -->
-* 
+* Đỗ Trường Giang
```
Comment (trGiang99) [inline chapter_optimization/adagrad_vn.md:447]:
```suggestion
* Đỗ Trường Giang
* Nguyễn Văn Quang
```
